prime_numbers.go

```diff
@@ -131,28 +131,36 @@ func getPrimesUpTo(n int) []int {
 // Send a boolean to the Done channel when finished.
 // The prime sieve: Daisy-chain Filter processes.
 func GetPrimeNumbers() (<-chan int, chan<- bool) {
-	ch := make(chan int) // Create a new channel.
+	ch := make(chan int) // Initial channel to send all the numbers on.
 	ctx, cancel := context.WithCancel(context.Background())
-	go generate(ctx, ch) // Launch Generate goroutine.
 
-	primeCh := make(chan int) // Create return channel.
-	doneCh := make(chan bool) // Create done channel.
+	go generate(ctx, ch) // Launch Generate goroutine to send all the numbers.
+
+	primeCh := make(chan int) // Create main prime return channel.
+	doneCh := make(chan bool) // Create the done channel.
```

Comment:
Instead of a doneCh, can you take a context as a parameter and have the caller cancel it when they are done?

Reply:
Good idea, thanks!